src/Support/ChainableClosure.php

```diff
@@ -15,31 +15,37 @@ final class ChainableClosure
     /**
      * Calls the given `$closure` when the given condition is true, "bound" to the same object.
      */
-    public static function boundWhen(Closure $condition, Closure $next): Closure
+    public static function boundWhen(object $self, Closure $condition, Closure $next): Closure
```

Comment:
I'm not sure why `$this` is used in a static, I think @nunomaduro might know, however these changes would be breaking changes if anyone is using these directly.